artiq/master/worker_db.py

```diff
@@ -119,6 +119,10 @@ def set(self, key, value, broadcast=False, persist=False, archive=True):
         if persist:
             broadcast = True
 
+        if not (broadcast or archive):
+            logger.warning("No dataset will be stored. Both 'broadcast' and 'archive' are set to False.",
```

Comment:
Fix the wording. "No dataset will be stored" means that there will not be any dataset stored.
Also no need to print the trace and the logging message is missing the info which dataset is affected.